lnwallet/channel.go

```diff
@@ -6715,12 +6715,19 @@ func (lc *LightningChannel) validateFeeRate(feePerKw chainfee.SatPerKWeight) err
 	// We'll ensure that we can accommodate this new fee change, yet still
 	// be above our reserve balance. Otherwise, we'll reject the fee
 	// update.
-	availableBalance, txWeight := lc.availableBalance()
-	oldFee := lnwire.NewMSatFromSatoshis(lc.localCommitChain.tip().fee)
+	_, txWeight := lc.availableBalance()
+
+	localBalance := lnwire.NewMSatFromSatoshis(
+		lc.channelState.LocalCommitment.LocalBalance.ToSatoshis(),
```

Comment:
ok sure, but then i think that we should also then use `availableBalance` in the `MaxFeeRate` function so that it also takes the pending htlc's into account. Otherwise `validateFeeRate` can not reliably be used to test the edge cases returned by `MaxFeeRate`.  Furthermore, I think that if we do use the `availableBalance` function in both MaxFeeRate and validateFeeRate, then we can remove the reserve balance checks in both since `availableBalance` already handles this for us